### ISSUE TITLE
..

### ISSUE THREAD

**Issue by cryptodescriptor** [hashcat/hashcat#2665]

**Comment by matrix** [comment]
Hi,

you need also change this from "6" to "0"

https://github.com/hashcat/hashcat/blob/master/src/modules/module_01300.c#L17

and you can use -O, works only with pure kernel.

**Comment by cryptodescriptor** [comment]
Changing the 6 to 0 did something, because it cracks when the cracked full (56 char) hash is in the potfile. But when it isn't present in the potfile, I get "Aborting session due to kernel self-test failure."

When bypassed with --self-test-disable, it fails to crack.

Appreciate the fast response by the way.

**Comment by matrix** [comment]
> Changing the 6 to 0 did something, because it finds the match when the full (56 char) hash is in the potfile. But when it isn't present in the potfile, I get "Aborting session due to kernel self-test failure."
> 
> When bypassed with --self-test-disable, it still fails to crack.
> 
> Appreciate the fast response by the way.

I do a simple test, and it works. Share you diff please.

**Comment by cryptodescriptor** [comment]
I have just noticted it works when I do not use -O.

**Comment by cryptodescriptor** [comment]
Quick question again.. Sorry

If I wanted to do this for all these modules:
1300 - SHA-224
1400 - SHA-256
10800 - SHA-384
1700 - SHA-512

17300 - SHA3-224
17400 - SHA3-256
17500 - SHA3-384
17600 - SHA3-512

with the same truncation size, how would I go about doing that? I would think it would be different each time.

For example.. This is how it is configured for -m 1400 (SHA2-256):
`static const u32   DGST_POS0      = 3;
static const u32   DGST_POS1      = 7;
static const u32   DGST_POS2      = 2;
static const u32   DGST_POS3      = 6;`

Changing the 6 wont do anything, because DGST_POS1=7 probably is not contained in the truncated version of the full hash.

So I tried changing it to this:
`static const u32   DGST_POS0      = 3;
static const u32   DGST_POS1      = 0;
static const u32   DGST_POS2      = 2;
static const u32   DGST_POS3      = 0`

But that did not work.

I also tried this:
`static const u32   DGST_POS0      = 3;
static const u32   DGST_POS1      = 0;
static const u32   DGST_POS2      = 0;
static const u32   DGST_POS3      = 0`

No avail.. I wish there were a truncation flag for the cmd line, but maybe it is complicated to calculate and would distrupt too many modules.
